### PR TITLE
Create update candidate for S1 and S0

### PR DESCRIPTION
**Pull request by hakonfam**
Sign both S0 and S1 (which is already signed by B0) in order to prepare update candidates for both slots.

### CODE REVIEW

**Comment by hakonfam** [comment]
cc @oyvindronningstad @sigvartmh

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -43,52 +106,73 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
...
     mcuboot_primary_app_PM_TARGET
     mcuboot_sign_target
     )
+
+  if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+    # Secure Boot (B0) is enabled, and we have to build update candidates
+    # for both S1 and S0.
+
+    foreach(slot s0_image;s1_image)
+      string(SUBSTRING ${slot} 0 2 slot_only) # 's0_image' -> 's0'
```
Comment (oyvindronningstad) [inline zephyr/CMakeLists.txt]: How about `foreach(slot s0;s1)` instead? In that case, `${slot}` -> `${slot}_image` and `${slot_only}` -> `${slot}`.

File: zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -43,52 +106,73 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
...
+      # Depend on both the target for the hex file, and the hex file itself.
+      set(dependencies "${${slot}_target};${${slot}_hex}")
+
+      set(out_path ${PROJECT_BINARY_DIR}/signed_by_mcuboot_and_b0_${slot})
+
+      sign(${${slot}_hex} # Hex file to sign
+        ${out_path}
+        ${slot_offset}
+        "${dependencies}"    # Need "..." to make it a list.
+        b_one_signed_hex     # Created file variable
```
Comment (oyvindronningstad) [inline zephyr/CMakeLists.txt]: b_one == b1? IMO "signed_hex" is sufficient.

File: zephyr/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,75 @@
 if(CONFIG_BOOTLOADER_MCUBOOT)
-  # Build a second bootloader image
+  function(sign to_sign_hex output_prefix offset sign_depends signed_hex_out)
```
Comment (oyvindronningstad) [inline zephyr/CMakeLists.txt]: Could this function be upstreamed?

File: zephyr/CMakeLists.txt [hunk 4]
```diff
@@ -9,28 +78,22 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
...
-
   get_property(app_binary_dir GLOBAL PROPERTY PROJECT_BINARY_DIR)
   set(merged_hex_file
     ${app_binary_dir}/mcuboot_primary_app.hex)
   set(merged_hex_file_depends
     mcuboot_primary_app_hex$<SEMICOLON>${PROJECT_BINARY_DIR}/mcuboot_primary_app.hex)
   set(sign_merged
     $<TARGET_EXISTS:partition_manager>)
-  set(to_sign_hex
+  set(app_to_sign_hex
```
[the remark this answers is not in the document]
Comment (hakonfam) [reply zephyr/CMakeLists.txt]: fixed

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (hakonfam) [reply zephyr/CMakeLists.txt]: I see your point. But after making the change, I find it harder to see the relation between `${slot}` and `${slot}_image`. 

Earlier you could easily see if the variable being used was that of the full Sx slot, or just the image. I agree that the distinction is not clear to everyone, but that could be fixed using some docs.

File: zephyr/CMakeLists.txt (hunk 2 above)
Comment (hakonfam) [reply zephyr/CMakeLists.txt]: fixed

File: zephyr/CMakeLists.txt (hunk 3 above)
Comment (hakonfam) [reply zephyr/CMakeLists.txt]: Without multi image I don't see any benefit of this, as the user would have to manually point to what hex file should be signed etc.

Also, upstream doesn't have mcuboot as a west-module, hence this file does not exist upstream.

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (oyvindronningstad) [reply zephyr/CMakeLists.txt]: Ok, how about this?

    foreach(slot_only s0;s1)
        set(slot ${slot_only}_image)

Comment (hakonfam) [reply zephyr/CMakeLists.txt]: I agree

Comment (oyvindronningstad) [reply zephyr/CMakeLists.txt]: Ok, good :). My suggestion:

    foreach(parent_slot s0;s1)
        set(slot ${parent_slot}_image)

**Pull request merged**